src/solvers/refinement/string_refinement.cpp

```diff
@@ -689,6 +689,42 @@ void output_equations(
            << " == " << from_expr(ns, "", equations[i].rhs()) << std::endl;
 }
 
+/// Make character array accesses at index greater than max_index return
```

Comment:
What are the consequences of this? Do you detect at a later stage that such an access has happened?

Reply:
This is like constraining the arrays to have all the same character after some index, so if the formula we constructed (with this replaced array accesses) is satisfiable then there is a model for the original formula. Note that the model may be a bit different, for instance the underlying solver can come up with "abcde" but the correct model for the string solver is actually "ab???" (string-max-length = 2).

Reply:
Would it be safer to introduce an assumption that the length is within bounds instead?

Reply:
this is what I'm trying to avoid